pennylane/tape/tape.py

```diff
@@ -55,6 +55,49 @@ def get_active_tape():
     return QueuingManager.active_context()
 
 
+def _err_msg_for_some_meas_not_qwc(measurements):
+    """Error message for the case when some operators measured on the same wire are not qubit-wise commuting."""
+    msg = (
+        "Only observables that are qubit-wise commuting "
+        "Pauli words can be returned on the same wire, "
+        f"some of the following measurements do not commute:\n{measurements}"
+    )
+    return msg
+
+
+def _validate_computational_basis_sampling(measurements):
+    """Auxiliary function for validating computational basis state sampling with other measurements considering the
+    qubit-wise commutativity relation."""
+    ordinary_obs = []
+    comp_basis_obs = []
+    for o in measurements:
+        if o.samples_computational_basis:
+            comp_basis_obs.append(o)
+        else:
+            ordinary_obs.append(o)
+
+    if ordinary_obs:
+        for cb_obs in comp_basis_obs:
+            wires = (
+                cb_obs.wires
+                if cb_obs.wires != qml.wires.Wires([])
+                else qml.wires.Wires.all_wires([m.wires for m in measurements])
+            )
+            with QueuingManager.stop_recording():  # stop recording operations - the constructed operator is just aux
+                pauliz_for_cb_obs = (
+                    qml.PauliZ(wires)
+                    if len(wires) == 1
+                    else qml.operation.Tensor(*[qml.PauliZ(w) for w in wires])
+                )
+
+            for obs in ordinary_obs:
+                # Cover e.g., qml.probs(wires=wires) case by checking obs attr
+                if obs.obs is not None and not qml.grouping.utils.are_pauli_words_qwc(
+                    [obs.obs, pauliz_for_cb_obs]
+                ):
```

Comment:
I am not fully convinced that you need two for loops here. Since we are going with qubit-wise commutativity instead of just commutativity, can we just create a single `pauliz_for_cb_obs` which acts on the union of wires for each `cb_obs` in `comp_basis_obs`? Then just check qwc between that resulting operator and each of the other `ordinary_obs`?

Reply:
@Jaybsoni it was addressed in https://github.com/PennyLaneAI/pennylane/pull/3207/commits/1e36bb129dcefaac48b52966242a63fb2dd46a49. Would there be something further to improve?

Reply:
Ahh Amazing 👍🏼